### PR TITLE
[18895] GUI skeleton

### PR DESCRIPTION
**Pull request by JesusPoderoso**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+configure_project_cpp()
+
+###############################################################################
+# Load external projects
+###############################################################################
+set(QT_PATH "" CACHE STRING "Path to the Qt installation folder")
+list(APPEND CMAKE_PREFIX_PATH ${QT_PATH})
+
+find_package(Qt5 5.15 COMPONENTS Core Widgets Gui Qml Quick Charts QuickControls2 REQUIRED)
+find_package(fastcdr REQUIRED)
```
Comment (rsanchez15) [inline CMakeLists.txt:70]: I think you don't need all of this if you use `cmake_utils` to build the project

File: package.xml [hunk 2]
```diff
@@ -0,0 +1,34 @@
+<?xml version="1.0"?>
+<?xml-model href="http://download.ros.org/schema/package_format3.xsd" schematypens="http://www.w3.org/2001/XMLSchema"?>
+<package format="3">
+  <name>sustainml_frontend</name>
+  <!-->There must be a release. Update version accordingly<-->
+  <version>0.0.1</version>
```
Comment (rsanchez15) [inline package.xml:6]:
```suggestion
  <version>0.1.0</version>
```

File: package.xml [hunk 3]
```diff
@@ -0,0 +1,34 @@
+<?xml version="1.0"?>
+<?xml-model href="http://download.ros.org/schema/package_format3.xsd" schematypens="http://www.w3.org/2001/XMLSchema"?>
+<package format="3">
+  <name>sustainml_frontend</name>
```
Comment (rsanchez15) [inline package.xml:4]: I'd rename it to `sustainml_dashboard` or just `sustainml`. I'd prefer just `sustainml`

```suggestion
  <name>sustainml_dashboard</name>
```

File: CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,227 @@
...
+# - Configure log depending on LOG_INFO flag and CMake type
+configure_project_cpp()
+
+###############################################################################
+# Load external projects
+###############################################################################
+set(QT_PATH "" CACHE STRING "Path to the Qt installation folder")
+list(APPEND CMAKE_PREFIX_PATH ${QT_PATH})
+
+find_package(Qt5 5.15 COMPONENTS Core Widgets Gui Qml Quick Charts QuickControls2 REQUIRED)
```
Comment (rsanchez15) [inline CMakeLists.txt:69]: Remove `Charts` as we are not using them so far

File: src/qml/components/SML_Text.qml [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+    enum Text_kind
+    {
+        App_name,
+        Header_1,
+        Header_2,
+        Header_3,
+        Body
+    }
+    // External properties
+    property int text_kind: SML_Text.Text_kind.Body // Body kind as default
```
Comment (rsanchez15) [inline src/qml/components/SML_Text.qml:20]: I'd not combine Snake Case with Camel Case. Rename it to `SmlText`

**Review by rsanchez15 — CHANGES_REQUESTED** [review]

File: CMakeLists.txt (hunk 1 above)
Comment (JesusPoderoso) [reply CMakeLists.txt:70]: This will be addressed in other PR

**Review by rsanchez15 — APPROVED** [review]

**Pull request merged**